### PR TITLE
Duration converters

### PR DESCRIPTION
**Pull request by johanandren**
Refs #85

### CODE REVIEW

File: src/main/scala/scala/compat/java8/DurationConverters.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   * @throws IllegalArgumentException If the given Java Duration is out of bounds of what can be expressed with the
+   *                                  Scala Durations.
+   */
+  final def toScala(duration: java.time.Duration): scala.concurrent.duration.Duration = {
+    if (duration.getNano == 0) {
+      if (duration.getSeconds == 0) ScalaDuration.Zero
+      else FiniteDuration(duration.getSeconds, TimeUnit.SECONDS)
+    } else {
+      FiniteDuration(
+        duration.getSeconds * 1000000000 + duration.getNano,
```
Comment (johanandren) [inline src/main/scala/scala/compat/java8/DurationConverters.scala]: Should probably be a `Math.multiplyExact` here though and an overflow check after adding the nano part.

**Comment by jbgi** [comment]
any update on this one? @johanandren maybe it could be integrated into https://github.com/johanandren/timeforscala if not here?

**Comment by SethTisue** [comment]
not sure why/how this fell through the cracks...

@johanandren could you update this to add a section to README.md? then we can merge, I think

File: src/main/scala/scala/compat/java8/DurationConverters.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * Transform a Java duration into a Scala duration. If the nanosecond part of the Java duration is zero the returned
+   * duration will have a time unit of seconds and if there is a nanoseconds part the Scala duration will have a time
+   * unit of nanoseconds.
+   *
+   * @throws IllegalArgumentException If the given Java Duration is out of bounds of what can be expressed with the
+   *                                  Scala Durations.
+   */
+  final def toScala(duration: java.time.Duration): scala.concurrent.duration.Duration = {
```
Comment (jbgi) [inline src/main/scala/scala/compat/java8/DurationConverters.scala]: I think this would be more useful if it return a `FiniteDuration`.

**Comment by johanandren** [comment]
I'll get this done soon, have been having sbt issues today though, so couldn't verify my changes.

**Comment by SethTisue** [comment]
the CI failure is #97 — unrelated to the changes in this PR.

**Comment by huntc** [comment]
So... here I was looking at the doc - 15 minutes later I discover that the doc its wrong. :-) I'd like to see a release soon if at possible - thanks.

**Comment by johanandren** [comment]
It isn't urgent for me, but it seems others are interested :)

**Comment by SethTisue** [comment]
@huntc okay I'll roll a release and report here when it's done. probably today

**Comment by huntc** [comment]
Hi @SethTisue - thanks so much. Really appreciated.

**Comment by SethTisue** [comment]
0.9.0 (for 2.11 and 2.12 only; skipping 2.13 this time because of #97) is on its way to Maven Central

**Comment by huntc** [comment]
Thanks!

**Pull request merged**